packages/expo-cli/src/credentials/api/ios.ts

```diff
@@ -349,35 +233,226 @@ export class IosApi {
     ]) as appleApi.ProvisioningProfile;
   }
 
-  async _deleteProvisioningProfileApi(appLookupParameters: AppLookupParameters): Promise<void> {
-    await this.api.postAsync(`credentials/ios/provisioningProfile/delete`, appLookupParameters);
+  public async updateProvisioningProfile(
+    appLookupParams: AppLookupParams,
+    provisioningProfile: appleApi.ProvisioningProfile
+  ): Promise<appleApi.ProvisioningProfile> {
+    const appCredentialsIndex = this.getAppCredentialsCacheIndex(appLookupParams);
+    const { accountName } = appLookupParams;
```

Comment:
I feel like the complexity of the cache has gotten to the point where it makes sense to abstract the details away and put it in it's own class. 

If we had  a cache class where we could call methods like `get(appLookupParams, PROVISIONING_PROFILE)` or `set(appLookupParams, PROVISIONING_PROFILE)`, the code would be simpler to follow.

Reply:
If this was long term solution, I agree, but I'm not sure if it makes sense to extract that if it will be replaced by graphql caching